pyecoregen/ecore.py

```diff
@@ -208,6 +235,7 @@ def filter_pyfqn(cls, value, relative_to=0):
         Args:
             relative_to: If greater 0, the returned path is relative to the first n directories.
         """
+        name_translation = {'ecore': 'pyecore.ecore'}
```

Comment:
Please adapt name or add small comment, why names have to be translated. Do you expect this to grow in the future? Could this be derived somehow from attributes in `pyecore` directly? I'd like to delegate this setting to `pyecore`if possible at all.

Reply:
Here, the translation is very specific to the code generation. Currently, the package name is used for the generation and the reference to external metamodel should use this name. However, in very specific cases, the module name should be changed. This is typically the cas for Ecore, the module name is not simply `ecore`, but `pyecore.ecore`. This list could grow if other metamodel implementation are generated in `pyecore` (e.g: XMLTypes). 

I get your point, so I see two solutions:

1. a dedicated method is called from `pyecore.utils` that take the `EObject` as parameter and returns the right import,
2. the `name_translation` is renamed `package_name_translation`  and set as class instance attribute, so it can be 'extended' if needed in specific generations.

I would prefer the second solution as this translation is really dependent on `pyecoregen` and the metamodel Python code generation, but I can make something in `pyecore` if you think this is more related to `pyecore`.

Reply:
I suggested `pyecore` since the package structure `pyecore.ecore` results from there. But I am fine with (2) as well. I'd suggest a name like `module_path_map` and a corresponding comment like "Maps name of specific Ecore packages to their corresponding Python module path."